### PR TITLE
Feature: MySQL port option

### PR DESCRIPTION
**Pull request by Maaiins**
Ref #364

### CODE REVIEW

**Comment by padams** [comment]
Something is going on with your diffs. They show to me that the whole files is changed.

**Comment by Maaiins** [comment]
Yeah that´s true. I will investigate it.... sorry

**Comment by Maaiins** [comment]
I don´t know why. Is it ok when I mark the specific change with comments?

File: modules/base/installConfig.php [hunk 1]
```diff
@@ -1,147 +1,157 @@
...
+		$v3->setValues($this->getParam('db_user'));
+		$v3->setErrorMessage("Database user is required.");
+		$this->setValidation('db_user', $v3);
+		
+		$v4 = owa_coreAPI::validationFactory('required');
+		$v4->setValues($this->getParam('db_password'));
+		$v4->setErrorMessage("Database password is required.");
+		$this->setValidation('db_password', $v4);
+
+        $v5 = owa_coreAPI::validationFactory('required');
```
Comment (Maaiins) [inline modules/base/installConfig.php]: Added Validation for port

**Comment by Maaiins** [comment]
Commit looks correct now

File: modules/base/installConfig.php [hunk 2]
```diff
@@ -44,6 +44,11 @@ function __construct($params) {
 		$v1->setValues($this->getParam('db_host'));
 		$v1->setErrorMessage("Database host is required.");
 		$this->setValidation('db_host', $v1);
+
+        $v8 = owa_coreAPI::validationFactory('required');
```
Comment (padams) [inline modules/base/installConfig.php]: Do you think this should be required? Seems like it's really an optional setting as we have a stable default for it.

File: modules/base/templates/install_config_entry.php [hunk 3]
```diff
@@ -32,6 +32,14 @@
 			</span>
 			<span class="form-instructions">This is the host that your database resides on. Localhost is ok.</span>
 		</p>
+
+        <p class="form-row">
+            <span class="form-label">Database Port:</span>
+            <span class="form-field">
+				<input type="text"size="30" name="<?php echo $this->getNs();?>db_port" value="<?php echo $config['db_port'];?>">
+			</span>
+            <span class="form-instructions">This is your database Port. Normally 3306</span>
```
Comment (padams) [inline modules/base/templates/install_config_entry.php]: I think this should be optional. Maybe we can say something like: "This optional setting is used to change the port for your database. OWA will use the default port 3306 if you leave this empty."

**Pull request merged**